src/Nethermind/Nethermind.Api/Factories/IApiComponentFactory.cs

```diff
@@ -0,0 +1,6 @@
+namespace Nethermind.Api.Factories;
+
+public interface IApiComponentFactory<out T>
```

Comment:
Shouldnt the DI already have mechanism for factory, meaning this type is unnecessary? Like [this](https://learn.microsoft.com/en-us/dotnet/api/microsoft.extensions.dependencyinjection.servicecollectionserviceextensions.addsingleton?view=dotnet-plat-ext-7.0#microsoft-extensions-dependencyinjection-servicecollectionserviceextensions-addsingleton(microsoft-extensions-dependencyinjection-iservicecollection-system-type-system-func((system-iserviceprovider-system-object))))

Reply:
Yes, i mentioned that in one of my comments